src/hotspot/share/runtime/globals.hpp

```diff
@@ -1999,6 +1999,13 @@ const int ObjectAlignmentInBytes = 8;
           "more eagerly at the cost of higher overhead. A value of 0 "      \
           "(default) disables native heap trimming.")                       \
           range(0, UINT_MAX)                                                \
+                                                                            \
+  product(uint, SecondarySuperMissBackoff, 1000, EXPERIMENTAL,              \
```

Comment:
I don't think the discriminating factor for calling an option "diagnostic" or "experimental" is its default value. Rather it is its target use.

As per `globals.hpp`:

```
// DIAGNOSTIC options are not meant for VM tuning or for product modes.
//    They are to be used for VM quality assurance or field diagnosis
//    of VM bugs.

// EXPERIMENTAL flags are in support of features that may not be
//    an officially supported part of a product, but may be available
//    for experimenting with. They could, for example, be performance
//    features that may not have undergone full or rigorous QA, but which may
//    help performance in some cases and released for experimentation
//    by the community of users and developers. 
```

...and this one is obviously the experimental performance feature.

Reply:
I referred to default value because in your particular case it controls whether the feature is turned on or off. Calling a feature experimental when it is unconditionally turned on in product builds looks a bit weird, doesn't it?

Reply:
Well, maybe :) One might ask another question: if I am experimenting with the value of this option, by switching from non-zero default to another non-zero value for performance evaluation in the field, is it really "diagnostic"? I think "experimental" captures the whole thing better. But I can change to diagnostic, if you insist.

Reply:
A better question to ask is "Why do I have to specify -XX:+UnlockExperimentalVMOptions to turn the feature OFF?". Diagnosing a performance anomaly may involve both turning the logic off or increasing the limit. So, I do think that diagnostic is a better fit here.

Reply:
Another question: should the flag be platform-specific? I'd expect different platforms may eventually settle on different defaults (e.g., platforms w/o support should set it to 0 right away) and, also, it would be better to print warnings when users try to explicitly   set the flag to non-zero value when proper support is missing.

Reply:
All right, diagnostic platform-dependent it is.